connexion/spec.py

```diff
@@ -158,6 +160,17 @@ def from_file(cls, spec, *, arguments=None, base_uri=""):
         spec = cls._load_spec_from_file(arguments, specification_path)
         return cls.from_dict(spec, base_uri=base_uri)
 
+    @classmethod
+    def from_url(cls, spec, *, arguments=None, base_uri=""):
+        """
+        Takes in a path to a YAML file, and returns a Specification
+        """
+        spec_content = requests.get(spec).content
+        with tempfile.NamedTemporaryFile() as tfile:
+            tfile.write(spec_content)
+            tfile.flush()
+            return cls.from_file(tfile.name, arguments=arguments, base_uri=base_uri)
```

Comment:
Could we reuse the code we already use to resolve remote references?

https://github.com/spec-first/connexion/blob/5de6dccfbb6a0ed39966dcff5038e7708c93d550/connexion/json_schema.py#L56-L62

I like that it raises a clear error for non-successful requests and doesn't use a tempfile.